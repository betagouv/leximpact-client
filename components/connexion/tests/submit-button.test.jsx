import { createShallow } from "@material-ui/core/test-utils";

import SubmitButton from "../submit-button";

describe("components | connexion | submit-button", () => {
  let shallow;
  beforeAll(() => {
    shallow = createShallow();
  });
  describe("snapshot", () => {
    it("correspond a l'etat normal", () => {
      const props = {
        disabled: false,
        isLoading: false,
      };
      const wrapper = shallow(<SubmitButton {...props} />);
      expect(wrapper).toBeDefined();
      expect(wrapper).toMatchSnapshot();
    });

    it("correspond a l'etat disabled", () => {
      const props = {
        disabled: true,
        isLoading: false,
      };
      const wrapper = shallow(<SubmitButton {...props} />);
      expect(wrapper).toBeDefined();
      expect(wrapper).toMatchSnapshot();
    });

    it("correspond a l'etat loading", () => {
      const props = {
        disabled: false,
        isLoading: true,
      };
      const wrapper = shallow(<SubmitButton {...props} />);
      expect(wrapper).toBeDefined();
      expect(wrapper).toMatchSnapshot();
    });
  });
});
