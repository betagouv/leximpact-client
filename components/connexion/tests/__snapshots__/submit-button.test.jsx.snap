// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components | connexion | submit-button snapshot correspond a l'etat disabled 1`] = `
<SubmitButton
  classes={
    Object {
      "circularProgress": "SubmitButton-circularProgress-3",
      "container": "SubmitButton-container-1",
      "fab": "SubmitButton-fab-2",
    }
  }
  disabled={true}
  isLoading={false}
/>
`;

exports[`components | connexion | submit-button snapshot correspond a l'etat loading 1`] = `
<SubmitButton
  classes={
    Object {
      "circularProgress": "SubmitButton-circularProgress-3",
      "container": "SubmitButton-container-1",
      "fab": "SubmitButton-fab-2",
    }
  }
  disabled={false}
  isLoading={true}
/>
`;

exports[`components | connexion | submit-button snapshot correspond a l'etat normal 1`] = `
<SubmitButton
  classes={
    Object {
      "circularProgress": "SubmitButton-circularProgress-3",
      "container": "SubmitButton-container-1",
      "fab": "SubmitButton-fab-2",
    }
  }
  disabled={false}
  isLoading={false}
/>
`;
